test/yaml-rest-runner/src/test/java/org/elasticsearch/test/rest/yaml/section/SkipSectionTests.java

```diff
@@ -76,49 +113,33 @@ public void testParseSkipSectionVersionNoFeature() throws Exception {
             version:     " - %s"
             reason:      Delete ignores the parent param""", version));
 
-        SkipSection skipSection = SkipSection.parse(parser);
-        assertThat(skipSection, notNullValue());
-        assertThat(skipSection.getLowerVersion(), equalTo(VersionUtils.getFirstVersion()));
-        assertThat(skipSection.getUpperVersion(), equalTo(version));
-        assertThat(skipSection.getFeatures().size(), equalTo(0));
-        assertThat(skipSection.getReason(), equalTo("Delete ignores the parent param"));
-    }
-
-    public void testParseSkipSectionAllVersions() throws Exception {
-        parser = createParser(YamlXContent.yamlXContent, """
-            version:     " all "
-            reason:      Delete ignores the parent param""");
-
-        SkipSection skipSection = SkipSection.parse(parser);
-        assertThat(skipSection, notNullValue());
-        assertThat(skipSection.getLowerVersion(), equalTo(VersionUtils.getFirstVersion()));
-        assertThat(skipSection.getUpperVersion(), equalTo(Version.CURRENT));
-        assertThat(skipSection.getFeatures().size(), equalTo(0));
-        assertThat(skipSection.getReason(), equalTo("Delete ignores the parent param"));
+        var skipSectionBuilder = SkipSection.parseInternal(parser);
+        assertThat(skipSectionBuilder, notNullValue());
+        assertThat(skipSectionBuilder.version, not(emptyOrNullString()));
+        assertThat(skipSectionBuilder.testFeatures.size(), equalTo(0));
+        assertThat(skipSectionBuilder.reason, equalTo("Delete ignores the parent param"));
     }
 
     public void testParseSkipSectionFeatureNoVersion() throws Exception {
         parser = createParser(YamlXContent.yamlXContent, "features:     regex");
 
-        SkipSection skipSection = SkipSection.parse(parser);
-        assertThat(skipSection, notNullValue());
-        assertThat(skipSection.isVersionCheck(), equalTo(false));
-        assertThat(skipSection.getFeatures().size(), equalTo(1));
-        assertThat(skipSection.getFeatures().get(0), equalTo("regex"));
-        assertThat(skipSection.getReason(), nullValue());
+        var skipSectionBuilder = SkipSection.parseInternal(parser);
+        assertThat(skipSectionBuilder, notNullValue());
+        assertThat(skipSectionBuilder.version, emptyOrNullString());
+        assertThat(skipSectionBuilder.testFeatures, hasSize(1));
+        assertThat(skipSectionBuilder.testFeatures, contains("regex"));
```

Comment:
nit - `contains` checks the collection _only_ contains those items, and no others, so the size check above it is redundant

Reply:
Good catch, will simplify it. Thanks!